axlearn/common/attention.py

```diff
@@ -3956,15 +3956,22 @@ def policy(prim, *_, **params):
     return policy
 
 
-SELF_ATTENTION_SAVE_PATTERN = ".*([qkvo]_proj|context)"
-FEED_FORWARD_SAVE_PATTERN = ".*linear[12]_.*"
+# Regex patterns for matching remat names
+class RematRegexSavePatterns(enum.Enum):
+    QKV_PROJ = r".*[kqv]_proj"
+    O_PROJ = r".*o_proj"
+    CONTEXT = r".*context"
+    LINEAR1_X = r".*linear1_[01]"
+    LINEAR2_X = r".*linear2_[01]"
+    SELF_ATTENTION = re.compile("|".join([QKV_PROJ, O_PROJ, CONTEXT])).pattern
```

Comment:
```suggestion
    SELF_ATTENTION = ".*([qkvo]_proj|context)"
```

We prefer keeping this unchanged to avoid updating golden config.

Reply:
Done